exporters/exporter_factory.go

```diff
@@ -41,3 +41,13 @@ func ExporterByName(name string, dataDir string, logger *logrus.Logger) (Exporte
 	}
 	return val, nil
 }
+
+// ExporterBuilderByName returns a Processor constructor for the name provided
+func ExporterBuilderByName(name string) (ExporterConstructor, error) {
```

Comment:
May as well pass in the config and call the constructor as well. That way the caller doesn't need to care about how things are actually made.

Reply:
There is a function above this that does that.  I wanted a function to pull out a constructor so that the pipeline can perform the same function all all objects in a defined sequence.  I figured it would be easier to reason about initialization that way.

Reply:
This suggestion applies to all of them, Use the `ByName` variant across each. There are probably a few good opportunities for generics in this code.